tf_agents/policies/tf_policy.py

```diff
@@ -239,7 +239,7 @@ def get_initial_state(self, batch_size):
     return self._get_initial_state(batch_size)
 
   def _maybe_reset_state(self, time_step, policy_state):
-    if policy_state is ():  # pylint: disable=literal-comparison
+    if policy_state == ():
```

Comment:
this will cause failures if policy_state is a Tensor, and the output of policy_state == () will be a bool tensor.  we dont want that here since it's mean tto be a check on the type of policy_state.

feel free to add a comment to this effect.